### PR TITLE
Added cleanup of temp files after Kilosort run

### PR DESCRIPTION
**Pull request by DradeAW**
Tested this: works for a local run but not when using container :/

Is this because I need to clear Singularity cache and rebuild?

### CODE REVIEW

File: src/spikeinterface/sorters/external/kilosortbase.py [hunk 1]
```diff
@@ -208,6 +208,16 @@ def _run_from_folder(cls, sorter_output_folder, params, verbose):
...
             raise Exception(f"{cls.sorter_name} returned a non-zero exit code")
 
+        # Clean-up temporary files
+        if params["delete_tmp_files"]:
+            for temp_file in sorter_output_folder.glob("*.m"):
+                temp_file.unlink()
+            for temp_file in sorter_output_folder.glob("*.mat"):
+                temp_file.unlink()
+            (sorter_output_folder / "temp_wh.dat").unlink()
+        if params["delete_recording_dat"]:
```
Comment (alejoe91) [inline src/spikeinterface/sorters/external/kilosortbase.py:218]: Here you should check if it exists. For binary recordings, we dont make a copy!

Comment (DradeAW) [reply src/spikeinterface/sorters/external/kilosortbase.py]: Done.

**Comment by DradeAW** [comment]
So after clearing the cache for singularity and trying again:

The temporary files are still not deletd, and if I explicitly set `ss.run_sorter("kilosort3", recording, "test/", delete_tmp_files=True, singularity_image=True)`, I get the following crash:

```
Traceback (most recent call last):
  File "/mnt/raid0/data/test_docker/test_docker.py", line 11, in <module>
    ss.run_sorter("kilosort3", recording, "test/", delete_tmp_files=True, singularity_image=True)
  File "/export/home1/users/nsr/wyngaard/dev/forked_repo/spikeinterface/src/spikeinterface/sorters/runsorter.py", line 135, in run_sorter
    return run_sorter_container(
           ^^^^^^^^^^^^^^^^^^^^^
  File "/export/home1/users/nsr/wyngaard/dev/forked_repo/spikeinterface/src/spikeinterface/sorters/runsorter.py", line 585, in run_sorter_container
    raise SpikeSortingError(f"Spike sorting in {mode} failed with the following error:\n{run_sorter_output}")
spikeinterface.sorters.utils.misc.SpikeSortingError: Spike sorting in singularity failed with the following error:
Traceback (most recent call last):
  File "/mnt/raid0/data/test_docker/in_container_sorter_script.py", line 17, in <module>
    sorting = run_sorter_local(
  File "/mnt/raid0/data/test_docker/in_container_python_base/lib/python3.8/site-packages/spikeinterface/sorters/runsorter.py", line 165, in run_sorter_local
    SorterClass.set_params_to_folder(recording, output_folder, sorter_params, verbose)
  File "/mnt/raid0/data/test_docker/in_container_python_base/lib/python3.8/site-packages/spikeinterface/sorters/basesorter.py", line 176, in set_params_to_folder
    raise AttributeError("Bad parameters: " + str(bad_params))
AttributeError: Bad parameters: ['delete_tmp_files']
```

The same line without `singularity_image=True` works without a crash, and with expected behaviour.

@alejoe91 any idea?

**Comment by alejoe91** [comment]
@DradeAW to use local changes in the container, just set this env variable:

```
export SPIKEINTERFACE_DEV_PATH="path-to-cloned-repo"
```

Then when running the container you should see: `Installing dev spikeinterface from local machine`

**Comment by DradeAW** [comment]
Ah you're right, I completely forgot about that!
This variable is set, but points to the main depo, not the forked one with the changes ^^

Then I guess this PR is ready to be merged (on my side at least)

**Comment by alejoe91** [comment]
Can you test that it works in the container before merging? :)

**Comment by DradeAW** [comment]
I can confirm it works!

**Review by alejoe91 — APPROVED** [review]

**Pull request merged**